linkerd/meshtls/rustls/src/creds/verify.rs

```diff
@@ -1,23 +1,33 @@
-use std::convert::TryFrom;
-use std::sync::Arc;
-use std::time::SystemTime;
+use std::{convert::TryFrom, sync::Arc};
 use tokio_rustls::rustls::{
     self,
-    client::{self, ServerCertVerified, ServerCertVerifier},
+    client::{
+        self,
+        danger::{ServerCertVerified, ServerCertVerifier},
+    },
+    crypto::WebPkiSupportedAlgorithms,
+    pki_types::{CertificateDer, ServerName, UnixTime},
     server::ParsedCertificate,
-    Certificate, RootCertStore, ServerName,
+    RootCertStore,
 };
 use tracing::trace;
 
-pub(crate) struct AnySanVerifier(Arc<RootCertStore>);
+#[derive(Debug)]
+pub(crate) struct AnySanVerifier {
+    roots: Arc<RootCertStore>,
+    supported: WebPkiSupportedAlgorithms,
+}
 
 impl AnySanVerifier {
     pub(crate) fn new(roots: impl Into<Arc<RootCertStore>>) -> Self {
-        Self(roots.into())
+        Self {
+            roots: roots.into(),
+            supported: rustls::crypto::ring::default_provider().signature_verification_algorithms,
+        }
     }
 }
 
-// This is derived from `rustls::client::WebPkiVerifier`.
+// This is derived from `rustls::client::WebPkiServerVerifier`.
```

Comment:
We should probably update the link/copyright below this to refer to the appropriate code.

Reply:
Done